### DISCUSSION TITLE
What's the difference between pion/ice and ice code in pion/webrtc

### DISCUSSION THREAD

**Discussion by iPush** [pion/webrtc#2331] [Q&A] [answered]
Hi, i'm new here, Pion is a great work.
I'm wondering what's the difference between pion/ice and ice parts in pion/webrtc?

- pion/webrtc is the high-level wrap of pion/ice (and other stuff, stun/turn)
- pion/ice is the low-level and detailed implementation of ICE RFC
so , developer should just program with  pion/webrtc to use pion/ice ?

am i right ? 
best wishes

**Comment by Sean-Der** [comment, marked as answer]
Hi @iPush 

Thank you so much for your kind words! Hope you are enjoying using Pion :) Your explanation sounds right to me!

The `ICE` objects in `pion/webrtc` come from the [webrtc-pc](https://w3c.github.io/webrtc-pc/) spec. This defines things like [ICECandidate](https://w3c.github.io/webrtc-pc/#rtcicecandidate-interface). We can't put these objects in `pion/ice` because they are WebRTC specific.

`pion/ice` is also used by people that don't use WebRTC you can see the importers [here](https://pkg.go.dev/github.com/pion/ice/v2?tab=importedby). I try and make sure that people that don't care about WebRTC aren't forced to import it.

----

If you are building something with `pion/webrtc` I would try to avoid using `pion/ice` (if possible). It is a lower level interface and the API may change at a different time then `pion/webrtc`. If you are building something with ICE using `pion/ice` directly makes sense to me!

**Reply by iPush** [reply]
Thank you so much!
